mlflow/store/artifact/hdfs_artifact_repo.py

```diff
@@ -69,11 +74,22 @@ def log_artifacts(self, local_dir, artifact_path=None):
                 if not hdfs.exists(hdfs_subdir_path):
                     hdfs.mkdir(hdfs_subdir_path)
 
-                for each_file in files:
-                    source = os.path.join(subdir_path, each_file)
-                    destination = posixpath.join(hdfs_subdir_path, each_file)
-                    with open(source, "rb") as f:
-                        hdfs.upload(destination, f)
+                num_of_cpus = os.cpu_count() or _NUM_DEFAULT_CPUS
```

Comment:
Can we reuse the code in `ArtifactRepository.download_artifacts`?

https://github.com/mlflow/mlflow/blob/9972267786d895b42c1d7d5e4bdc07aad37bbb95/mlflow/store/artifact/artifact_repo.py#L108

Reply:
I guess, we can reuse the threadpool defined in ArtifactRepository. What do you think?